worker/buildbot_worker/test/util/misc.py

```diff
@@ -186,7 +188,16 @@ class StdoutAssertionsMixin(object):
     """
 
     def setUpStdoutAssertions(self):
-        self.stdout = cStringIO.StringIO()
+        #
+        # sys.stdout is implemented differently
+        # in Python 2 and Python 3, so we need to
+        # override it differently.
+        # In Python 2, sys.stdout is a byte stream.
+        # In Python 3, sys.stdout is a text stream.
+        if PY3:
+            self.stdout = StringIO()
+        else:
+            self.stdout = BytesIO()
```

Comment:
It looks weird, but the old StringIO and cStringIO modules cannot even be imported
in Python 3.  io.StringIO deals strictly with Unicode on Python 2, Python3,
and io.BytesIO deals strictly with bytes on Python 2, Python 3.

sys.stdout / sys.stderr are implemented very differently on Python 3 as I mentioned in the comments,
so in order to keep this code working on both Python 2 and Python 3,
using BytesIO on Python 2, and StringIO on Python 3 is necessary.
For this particular case of overriding sys.stdout, it cannot be avoided.